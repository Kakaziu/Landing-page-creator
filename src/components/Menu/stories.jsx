import Menu from '.';
import links from '../NavLink/mock';

export default {
  title: 'Menu',
  component: Menu,
  args: {
    links: links,
    logoData: {
      text: 'logo',
      srcImg: '',
      link: '#target',
    },
  },

  argTypes: {
    children: { type: 'string' },
  },
};

export const Template = (args) => (
  <div>
    <Menu {...args} />
  </div>
);
