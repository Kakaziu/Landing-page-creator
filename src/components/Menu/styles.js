import styled, { css } from 'styled-components';
import { Container as SectionContainer } from '../SectionContainer/styles';
import { Container as LogoLink } from '../LogoLink/styles';

const menuOnVisible = () => css`
  visibility: visible;
  opacity: 1;
`;

export const Container = styled.div`
  ${({ theme, visible }) => css`
  position: fixed;
  z-index: 5;
  padding: 10px 0px;
  width: 100%;
  background: ${theme.colors.white};
  transition: 0.3s;
  overflow-y: auto;
  border-bottom: 1px solid ${theme.colors.mediumGray};

  ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }

    @media ${theme.media.lteMedium} {
      top: 0;
      left: 0;
      right: 0;
      bottom: 0;
      height: 100vh;
      display: flex;
      justify-content: center;
      align-items: center;
      visibility: hidden;
      opacity: 0;
      ${visible === 1 && menuOnVisible()}

      ${LogoLink} {
        margin-bottom: 3rem;
        justify-content: center;
      }
    }
  `}
`;

export const Button = styled.button`
  ${({ theme }) => css`
    position: fixed;
    z-index: 6;
    top: 2rem;
    right: 2rem;
    background: transparent;
    border: none;
    cursor: pointer;
    display: none;

    @media ${theme.media.lteMedium} {
      display: block;
    }
  `}
`;

export const MenuContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: space-between;

  ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }

    @media ${theme.media.lteMedium} {
      display: block;
      height: 42rem;
      overflow-y: auto;
      padding: ${theme.spacings.large};
      width: 100%;

      &::-webkit-scrollbar {
        width: 5px
      }

      &::-webkit-scrollbar-thumb {
        background-color: ${theme.colors.primaryColor};
        border-radius: 50px;
      }
    }
  `}
`;
