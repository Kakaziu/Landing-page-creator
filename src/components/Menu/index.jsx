import P from 'prop-types';
import LogoLink from '../LogoLink';
import NavLink from '../NavLink';
import SectionContainer from '../SectionContainer';
import * as Styled from './styles';
import { MdClose, MdMenu } from 'react-icons/md';
import { useState } from 'react';
import { theme } from '../../styles/theme';

const Menu = ({ links = [], logoData }) => {
  const [menuVisible, setMenuVisible] = useState(false);
  const [scrollY, setScrollY] = useState(0);

  function onSetMenuVisible() {
    if (menuVisible) {
      window.scroll({
        top: scrollY,
        left: 0,
        behavior: 'instant',
      });
    } else {
      setScrollY(window.scrollY);
    }

    setMenuVisible(!menuVisible);
  }

  return (
    <>
      <Styled.Button onClick={onSetMenuVisible} aria-label="Open/Close Menu">
        {menuVisible ? (
          <MdClose
            size="35"
            color={theme.colors.secondaryColor}
            aria-label="Close Menu"
          />
        ) : (
          <MdMenu
            size="35"
            color={theme.colors.secondaryColor}
            aria-label="Open Menu"
          />
        )}
      </Styled.Button>
      <Styled.Container menuVisible={menuVisible}>
        <SectionContainer>
          <Styled.MenuContainer>
            <LogoLink {...logoData} />
            <NavLink links={links} />
          </Styled.MenuContainer>
        </SectionContainer>
      </Styled.Container>
    </>
  );
};

Menu.propTypes = {
  ...NavLink.propTypes,
  logoData: P.shape(LogoLink.propTypes).isRequired,
};

export default Menu;
