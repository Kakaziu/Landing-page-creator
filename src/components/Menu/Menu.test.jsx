import { render } from '@testing-library/react';
import Menu from '.';

describe('<Menu/>', () => {
  it('should you write something here', () => {
    render(<Menu />);
  });
});
