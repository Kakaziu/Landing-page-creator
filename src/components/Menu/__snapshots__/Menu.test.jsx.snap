// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu/> should not render links 1`] = `
.c7 {
  color: #0A1128;
  text-decoration: none;
  font-size: 2.4rem;
  display: flex;
  align-items: center;
}

.c7 >img {
  width: 20rem;
}

.c5 {
  color: #0A1128;
  text-transform: uppercase;
  font-size: 3.2rem;
}

.c8 {
  color: #0A1128;
  display: flex;
  flex-flow: row wrap;
}

.c3 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c1 {
  position: fixed;
  z-index: 5;
  padding: 10px 0px;
  width: 100%;
  background: #FFFFFF;
  transition: 0.3s;
  overflow-y: auto;
  border-bottom: 1px solid #DDDDDD;
}

.c1 .c2 {
  padding-top: 0;
  padding-bottom: 0;
}

.c0 {
  position: fixed;
  z-index: 6;
  top: 2rem;
  right: 2rem;
  background: transparent;
  border: none;
  cursor: pointer;
  display: none;
}

.c4 {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.c4 .c2 {
  padding-top: 0;
  padding-bottom: 0;
}

@media (max-width:768px) {
  .c8 {
    flex-flow: column nowrap;
    align-items: center;
    align-content: center;
  }
}

@media (max-width:768px) {
  .c1 {
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    height: 100vh;
    display: flex;
    justify-content: center;
    align-items: center;
    visibility: hidden;
    opacity: 0;
  }

  .c1 .c6 {
    margin-bottom: 3rem;
    justify-content: center;
  }
}

@media (max-width:768px) {
  .c0 {
    display: block;
  }
}

@media (max-width:768px) {
  .c4 {
    display: block;
    height: 42rem;
    overflow-y: auto;
    padding: 3.2rem;
    width: 100%;
  }

  .c4::-webkit-scrollbar {
    width: 5px;
  }

  .c4::-webkit-scrollbar-thumb {
    background-color: #0A1128;
    border-radius: 50px;
  }
}

<div>
  <button
    aria-label="Open/Close Menu"
    class="c0"
  >
    <svg
      aria-label="Open Menu"
      color="#dc143c"
      fill="currentColor"
      height="35"
      stroke="currentColor"
      stroke-width="0"
      style="color: rgb(220, 20, 60);"
      viewBox="0 0 24 24"
      width="35"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
  <div
    class="c1"
  >
    <div
      class="c2 c3"
    >
      <div
        class="c4"
      >
        <h1
          class="c5"
        >
          <a
            class="c6 c7"
            href="#target1"
          >
            <span>
              Logo
            </span>
          </a>
        </h1>
        <nav
          class="c8"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Menu/> should render logo and nav 1`] = `
.c7 {
  color: #0A1128;
  text-decoration: none;
  font-size: 2.4rem;
  display: flex;
  align-items: center;
}

.c7 >img {
  width: 20rem;
}

.c5 {
  color: #0A1128;
  text-transform: uppercase;
  font-size: 3.2rem;
}

.c8 {
  color: #0A1128;
  display: flex;
  flex-flow: row wrap;
}

.c9 {
  color: #0A1128;
  font-size: 1.6rem;
  text-decoration: none;
  display: block;
  position: relative;
  padding: 1.6rem;
}

.c9::after {
  content: '';
  position: absolute;
  bottom: 0.5rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #dc143c;
  transition: all 300ms ease-in-out;
}

.c9:hover::after {
  left: 25%;
  width: 50%;
}

.c3 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c1 {
  position: fixed;
  z-index: 5;
  padding: 10px 0px;
  width: 100%;
  background: #FFFFFF;
  transition: 0.3s;
  overflow-y: auto;
  border-bottom: 1px solid #DDDDDD;
}

.c1 .c2 {
  padding-top: 0;
  padding-bottom: 0;
}

.c0 {
  position: fixed;
  z-index: 6;
  top: 2rem;
  right: 2rem;
  background: transparent;
  border: none;
  cursor: pointer;
  display: none;
}

.c4 {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.c4 .c2 {
  padding-top: 0;
  padding-bottom: 0;
}

@media (max-width:768px) {
  .c8 {
    flex-flow: column nowrap;
    align-items: center;
    align-content: center;
  }
}

@media (max-width:768px) {
  .c1 {
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    height: 100vh;
    display: flex;
    justify-content: center;
    align-items: center;
    visibility: hidden;
    opacity: 0;
  }

  .c1 .c6 {
    margin-bottom: 3rem;
    justify-content: center;
  }
}

@media (max-width:768px) {
  .c0 {
    display: block;
  }
}

@media (max-width:768px) {
  .c4 {
    display: block;
    height: 42rem;
    overflow-y: auto;
    padding: 3.2rem;
    width: 100%;
  }

  .c4::-webkit-scrollbar {
    width: 5px;
  }

  .c4::-webkit-scrollbar-thumb {
    background-color: #0A1128;
    border-radius: 50px;
  }
}

<div>
  <button
    aria-label="Open/Close Menu"
    class="c0"
  >
    <svg
      aria-label="Open Menu"
      color="#dc143c"
      fill="currentColor"
      height="35"
      stroke="currentColor"
      stroke-width="0"
      style="color: rgb(220, 20, 60);"
      viewBox="0 0 24 24"
      width="35"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
  <div
    class="c1"
  >
    <div
      class="c2 c3"
    >
      <div
        class="c4"
      >
        <h1
          class="c5"
        >
          <a
            class="c6 c7"
            href="#target1"
          >
            <span>
              Logo
            </span>
          </a>
        </h1>
        <nav
          class="c8"
        >
          <a
            class="c9"
            href="#target1"
            target="_self"
          >
            Link 1
          </a>
          <a
            class="c9"
            href="#target2"
            target="_self"
          >
            Link 2
          </a>
          <a
            class="c9"
            href="#target3"
            target="_self"
          >
            Link 3
          </a>
          <a
            class="c9"
            href="#target4"
            target="_self"
          >
            Link 4
          </a>
          <a
            class="c9"
            href="#target5"
            target="_self"
          >
            Link 5
          </a>
          <a
            class="c9"
            href="#target6"
            target="_self"
          >
            Link 6
          </a>
          <a
            class="c9"
            href="#target7"
            target="_self"
          >
            Link 7
          </a>
          <a
            class="c9"
            href="#target8"
            target="_self"
          >
            Link 8
          </a>
          <a
            class="c9"
            href="#target9"
            target="_self"
          >
            Link 9
          </a>
          <a
            class="c9"
            href="#target10"
            target="_self"
          >
            Link 10
          </a>
        </nav>
      </div>
    </div>
  </div>
</div>
`;
